### PR TITLE
feat(test): implement unit tests for our current codebase

### LINKED ISSUE #35: Feature-Write-Basic-Unit-Test-with-Jest
**Description**
Currently test are written in PyWright, which should be an e2e test framework. Unit tests are also needed to protect our core functionalities.

**Solution**
Develop unit tests using Jest and integrate to CI workflow.

### PR DESCRIPTION
**Pull request by MitchelHsu**
### Description
Currently test are written in PyWright, which should be an e2e test framework. Unit tests are also needed to protect our core functionalities. Planned to implement in Jest.

### Related Issues
- Fixes #35 

### Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Other (please specify):

### Testing Details
Screenshot 2024-11-24 at 5 12 29 PM

### Checklist
- [x] Code follows the style guide.
- [x] I have performed a self-review of my code.
- [ ] Documentation has been updated (if necessary).
- [x] Tests have been added or updated.

### Additional Notes
- Also update CI pipeline to run with Jest

### CODE REVIEW

**Review by kenny-lfx — APPROVED** [review]
The test appears to be solid. Thank you for the effort!

File: eslint.config.mjs [hunk 1]
```diff
@@ -18,6 +18,14 @@ export default [
       },
     },
   },
+  {
+    files: ['**/*.test.js', '**/*.spec.js'],
```
Comment (rashi-bhansali) [inline eslint.config.mjs:22]: what's the difference between test.js and spec.js?

**Review by rashi-bhansali — APPROVED** [review]
Looks good to me! Mocks can be made better suited to our new cell-like structure in the future.

File: eslint.config.mjs (hunk 1 above)
Comment (MitchelHsu) [reply eslint.config.mjs:22]: might not need `*.spec.js` in our case, will remove. Thanks!

**Review by rashi-bhansali — APPROVED** [review]

**Pull request merged**